File: Assets/TextManager.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local Map = ____lualib.Map
local __TS__New = ____lualib.__TS__New
local __TS__Iterator = ____lualib.__TS__Iterator
local ____exports = {}
local ____Helpers = require("Helpers")
local isEmpty = ____Helpers.isEmpty
____exports.default = __TS__Class()
local TextManager = ____exports.default
TextManager.name = "TextManager"
function TextManager.prototype.____constructor(self)
    self.texts = __TS__New(Map)
end
function TextManager.prototype.drawText(self)
    local prevFont = love.graphics.getFont()
    for ____, font in __TS__Iterator(self.texts:values()) do
        font:printFont()
    end
    love.graphics.setColor(1, 1, 1, 1)
    if not isEmpty(prevFont) then
        love.graphics.setFont(prevFont)
    end
end
function TextManager.prototype.addText(self, id, font)
    self.texts:set(id, font)
end
function TextManager.prototype.getText(self, id)
    return self.texts:get(id)
end
function TextManager.prototype.hideText(self, id)
    self.texts:delete(id)
end
function TextManager.prototype.updateText(self, id, newText)
    local text = self.texts:get(id)
    if not isEmpty(text) then
        text.text = newText
    end
end
function TextManager.prototype.disableText(self, id)
    local text = self.texts:get(id)
    if not isEmpty(text) then
        text:setDisabled(true)
    end
end
function TextManager.prototype.enableText(self, id)
    local text = self.texts:get(id)
    if not isEmpty(text) then
        text:setDisabled(false)
    end
end
function TextManager.getDefaultFontFilepath(self)
    return "Assets/Fonts/Germania.ttf"
end
function TextManager.setDefaultFont(self)
    local mainFont = love.graphics.newFont(self:getDefaultFontFilepath())
    love.graphics.setFont(mainFont)
    return mainFont
end
return ____exports

File: Assets/TextManager.ts
import { isEmpty } from "Helpers"
import FontWithPosition from "./FontWithPosition"
import { Font } from "love.graphics"

export default class TextManager {
    texts: Map<string, FontWithPosition>

    constructor() {
        this.texts = new Map<string, FontWithPosition>()
    }

    drawText(): void {
        const prevFont = love.graphics.getFont()
        for (const font of this.texts.values()) {            
            font.printFont()
        }
        love.graphics.setColor(1, 1, 1, 1)
        // Restore previous font
        if (!isEmpty(prevFont)) {
            love.graphics.setFont(prevFont)
        }
    }

    addText(id: string, font: FontWithPosition): void {
        this.texts.set(id, font)
    }

    getText(id: string): FontWithPosition | undefined {
        return this.texts.get(id)
    }

    hideText(id: string): void {
        this.texts.delete(id)
    }

    updateText(id: string, newText: string): void {
        const text = this.texts.get(id)
        if (!isEmpty(text)) {
            text.text = newText
        }
    }

    static getDefaultFontFilepath(): string {
        return "Assets/Fonts/Germania.ttf"
    }

    static setDefaultFont(): Font {
        const mainFont = love.graphics.newFont(this.getDefaultFontFilepath())
        love.graphics.setFont(mainFont)
        return mainFont
    }
}